plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -163,14 +171,16 @@ func findNodesThatFit(
 		// Create filtered list with enough space to avoid growing it
 		// and allow assigning.
 		filtered = make([]*api.Node, len(nodes))
-		meta := predicates.PredicateMetadata(pod, nodeNameToInfo)
+
 		errs := []error{}
 
 		var predicateResultLock sync.Mutex
 		var filteredLen int32
 		checkNode := func(i int) {
 			nodeName := nodes[i].Name
-			fits, failedPredicates, err := podFitsOnNode(pod, meta, nodeNameToInfo[nodeName], predicateFuncs)
+			// pFactory takes a pod, node info, and creates a black-box type which can be used by the predicate.
+			metaDataProducer := pFactory(pod, nodeNameToInfo)
```

Comment:
this looks like an error on my end when i pushed it back up.  too bad i dont have a good unit test for this optimization.